### ISSUE TITLE
Add new DNS provider `azuredns` to support new Azure authentication methods

### ISSUE THREAD

**Issue by pchanvallon** [vancluever/terraform-provider-acme#322]
Hello,

My Pull Request on the lego project has just been merged : https://github.com/go-acme/lego/pull/1830
It will add a new DNS provider called `azuredns`. This DNS provider uses the new Azure Go SDK and add new ways of authentication. 
Moreover note that the previous `azure` one is now deprecated because it is based on https://github.com/Azure/go-autorest and it is now out of support.

Therefore when the new version of lego project will be released (v4.13), can you please add this new DNS provider ?

Thanks a lot.

**Comment by vancluever** [comment]
Folks waiting for this, just FYI we're close to release - I'm just working on trying to resolve an issue with some dependencies on go-plugin, which I'm going to try and reach out to see if they can be expedited early next week. Once that's done we can release.

Cheers!

**Comment by nzdjb** [comment]
Hi @vancluever, is there any more info you can give on the dependency issue and when it will be resolved? We're waiting on this for functionality that was merged into lego a month ago.

**Comment by vancluever** [comment]
@nzdjb sorry for the delay. I'm working on pushing through a release for go-plugin to resolve the dependency issues. Once this is done we should be clear for releasing.

Thanks!

**Comment by vancluever** [comment]
Hey folks, 2.16.0 should be out shortly with the new changes. Thanks!

**Comment by vancluever** [comment]
Forgot to close this after we released and everything looked good - closing it out now!
